### PR TITLE
Encounter tables

### PR DESCRIPTION
**Pull request by ajscimone**

### CODE REVIEW

File: src/components/Encounters.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React, { Component } from "react";
+import {getEncounter} from "../data/encounter-tables";
+
+
+class Encounters extends Component {
+
+    render() {
+      let encounter = getEncounter("Coastal", "low");
+      console.log(encounter)
```
Comment (mweik14) [inline src/components/Encounters.js]: Initially just console log of a coastal encounter?

File: src/data/encounter-tables.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import {randomArrayEntry} from "../util/randomArrayEntry"
+import EncounterData from "../data/xanthars-random-encounter-tables.json";
+
+function parseEncounter(encounterJSON){
+  if (encounterJSON.event) {
+    return {"event": encounterJSON.event}
+  } else if (encounterJSON.quantity && encounterJSON.monsters) {
```
Comment (mweik14) [inline src/data/encounter-tables.js]: I see so the possibilities for encounter types are events and monsters with the monsters always having a quantity.

File: src/data/encounter-tables.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  encouterRatingArray.forEach( (row) => {
+    let expandedEncounterSubTable = createTableEntryFromRange(row)
+    rollTable = rollTable.concat(expandedEncounterSubTable)
+  });
+  encounterMap.set(rating, rollTable)
+  return encounterMap
+}
+
+
+function getEncounter(environment, challengeRating){
```
Comment (mweik14) [inline src/data/encounter-tables.js]: Does getEncounter always return arctic if its passed coastal , low as arguments in Encounter.js or am I missing something?

File: src/components/Encounters.js (hunk 1 above)
Comment (ajscimone) [reply src/components/Encounters.js]: This should be gone now.

File: src/data/encounter-tables.js (hunk 2 above)
Comment (ajscimone) [reply src/data/encounter-tables.js]: Yep. Except the monsters have more monsters embedded in the monsters tag. I plan to convert them all to events, and just parse the string for die rolls.

File: src/data/encounter-tables.js (hunk 3 above)
Comment (ajscimone) [reply src/data/encounter-tables.js]: It was just not finished yet. Take a look at the code now.

**Comment by mweik14** [comment]
- 2 (1d4) mages with 3 (1d4) shield guardians
- Street performers putting on a puppet show, involving two puppets beating each other with sticks to the amusement of the gathered crowd
- 7 (1d8) invisible stalkers

This is pretty Awesome!  should be good to merge.

**Pull request merged**